physics/clm_lake.f90

```diff
@@ -0,0 +1,5607 @@
+    real(kind_lake) :: thm(lbc:ubc)            ! intermediate variable (forc_t+0.0098*forc_hgt_t)
+    real(kind_lake) :: thv(lbc:ubc)            ! virtual potential temperature (kelvin)
+    real(kind_lake) :: thvstar                 ! virtual potential temperature scaling parameter
+    real(kind_lake) :: tksur                   ! thermal conductivity of snow/soil (w/m/kelvin)
+    real(kind_lake) :: tsur                    ! top layer temperature
+    real(kind_lake) :: tstar                   ! temperature scaling parameter
+    real(kind_lake) :: um(lbp:ubp)             ! wind speed including the stablity effect [m/s]
+    real(kind_lake) :: ur(lbp:ubp)             ! wind speed at reference height [m/s]
+    real(kind_lake) :: ustar(lbp:ubp)          ! friction velocity [m/s]
+    real(kind_lake) :: wc                      ! convective velocity [m/s]
+    real(kind_lake) :: zeta                    ! dimensionless height used in Monin-Obukhov theory
+    real(kind_lake) :: zldis(lbp:ubp)          ! reference height "minus" zero displacement height [m]
+    real(kind_lake) :: displa(lbp:ubp)         ! displacement (always zero) [m]
+    !    real(kind_lake) :: z0mg(lbp:ubp)           ! roughness length over ground, momentum [m]
+    real(kind_lake) :: z0hg(lbp:ubp)           ! roughness length over ground, sensible heat [m]
+    real(kind_lake) :: z0qg(lbp:ubp)           ! roughness length over ground, latent heat [m]
+    real(kind_lake) :: u2m                     ! 2 m wind speed (m/s)
+    real(kind_lake) :: u10(1)         ! 10-m wind (m/s) (for dust model)
+    real(kind_lake) :: fv(1)          ! friction velocity (m/s) (for dust model)
+
+    real(kind_lake) :: fm(lbp:ubp)             ! needed for BGC only to diagnose 10m wind speed
+    real(kind_lake) :: bw                       ! partial density of water (ice + liquid)
+    real(kind_lake) :: t_grnd_temp              ! Used in surface flux correction over frozen ground
+    real(kind_lake) :: betaprime(lbc:ubc)       ! Effective beta: 1 for snow layers, beta(islak) otherwise
+    character*256 :: message 
+    ! tgs COARE
+    real(kind_lake) :: tc, visc, ren
+
+      ! This assumes all radiation is absorbed in the top snow layer and will need
+      ! to be changed for CLM 4.
+    !
+    ! Constants for lake temperature model
+    !
+    real(kind_lake), parameter :: beta(2) = & ! fraction solar rad absorbed at surface: depends on lake type
+         (/0.4_kind_lake, 0.4_kind_lake/)  ! (deep lake, shallow lake)
+    ! This is the energy absorbed at the lake surface if no snow.
+    !    data za  /0.6_kind_lake, 0.5_kind_lake/
+    !    data eta /0.1_kind_lake, 0.5_kind_lake/
+    !-----------------------------------------------------------------------
+
+    ! Begin calculations
+    
+    !dir$ concurrent
+    !cdir nodep
+    forc_th(1)  = forc_t(1) * (forc_psrf(1)/ forc_pbot(1))**(rair/cpair)
+    forc_vp(1)  = forc_q(1) * forc_pbot(1)/ (0.622 + 0.378 * forc_q(1))
```

Comment:
.622 and .378 are redefined physical constants.

Reply:
0.622 is ratio_of_dry_air_to_water_vapor_gas_constants (usually denoted by lowercase epsilon) and .378 is 1 - the same.